io/src/ply_io.cpp

```diff
@@ -1209,14 +1209,36 @@ pcl::PLYWriter::writeBinary (const std::string &file_name,
   }
 
   std::ofstream fs;
-  fs.open (file_name.c_str ());      // Open file
+  fs.open (file_name.c_str (),
+           std::ios::out | std::ios::binary |
+               std::ios::trunc); // Open file in binary mode and erase current contents
+                                 // if any
   if (!fs)
   {
-    PCL_ERROR ("[pcl::PLYWriter::writeBinary] Error during opening (%s)!\n", file_name.c_str ());
+    PCL_ERROR ("[pcl::PLYWriter::writeBinary] Error during opening (%s)!\n",
+               file_name.c_str ());
     return (-1);
   }
 
-  unsigned int nr_points  = cloud.width * cloud.height;
+  if (!(writeBinary (fs, cloud, origin, orientation, use_camera) == 0))
+  {
+    fs.close();
+    return -1;
+  }
+  fs.close();
+  return 0;
+}
+
+int
+pcl::PLYWriter::writeBinary (std::ostream& fs,
+                             const pcl::PCLPointCloud2& cloud,
+                             const Eigen::Vector4f& origin,
+                             const Eigen::Quaternionf& orientation,
+                             bool use_camera)
+{
+  fs.imbue(std::locale::classic());
```

Comment:
I assume this makes use of a default locale independent on the user settings? Do we do the same when we read? Does it even matter in binary mode?

Reply:
Unsure about this, but I found we are doing the same thing in pcl::PCDWriter::writeBinary as well as pcl::PLYWriter::generateHeader (albeit we are writing ASCII characters to an ostringstream there). Setting the locale does seem to matter even in binary mode (see https://stackoverflow.com/questions/1831369/writing-binary-files-using-c-does-the-default-locale-matter). I also imagine TEST (PCL, PLYReaderWriter) would catch an issue with the number format